soconda.sh

```diff
@@ -77,49 +94,56 @@ else
     echo "Using config \"${config}\""
 fi
 
-# The env root name, used for the name of the generated module file
-envroot=$(basename ${envname})
-
-# The full environment name, including the root and version.
-fullenv="${envname}_${version}"
-
 # The path to the selected config directory
 confdir="${scriptdir}/config/${config}"
 if [ ! -d "${confdir}" ]; then
     echo "Config dir \"${confdir}\" does not exist"
     exit 1
 fi
 
-# The optional module init for this config
-modinit="${confdir}/module_init"
 
-# Activate the base environment
 if [ -n "${base}" ]; then
     conda_dir="${base}"
+    # Initialize conda
+    source "${conda_dir}/etc/profile.d/conda.sh"
+    # Conda initialization script will create conda function,
+    # but micromamba will create micromamba function.
+    # Here create a conda_exec function that unify conda and micromamba
+    # installation.
+    conda_exec () { conda "$@" ; }
 else
     # User did not specify where to find it
-    if [ -z "$(which conda)" ]; then
-        # conda is not in the path
+    if [ -n "$CONDA_EXE" ]; then
+        echo "Find conda command at ${CONDA_EXE}"
+        conda_dir="$(dirname $(dirname $CONDA_EXE))"
+        # Initialize conda
+        eval "$("$CONDA_EXE" 'shell.bash' 'hook')"
+        conda_exec () { conda "$@" ; }
+    elif [ -n "$MAMBA_EXE" ]; then
+        echo "Find micromamba command at ${MAMBA_EXE}"
+        conda_dir="$MAMBA_ROOT_PREFIX"
+        # Initialize micromamba
+        eval "$("$MAMBA_EXE" shell hook --shell bash)"
+        conda_exec () { micromamba "$@" ; }
+    else
+        # Could not find conda or micromamba
         echo "You must either activate the conda base environment before"
         echo "running this script, or you must specify the path to the base"
         echo "install with the \"-b <path to base>\" option."
         exit 1
-    else
-        # Conda is in the path
-        conda_dir="$(dirname $(dirname $(which conda)))"
     fi
 fi
 
-# Make sure that conda is initialized
-source "${conda_dir}/etc/profile.d/conda.sh"
-conda activate base
 
 # Conda package cache.  In some cases this can get really big, so
 # we point it to a temp location unless the user has overridden it.
 if [ -z ${CONDA_PKGS_DIRS} ]; then
     if [ -z ${NERSC_HOST} ]; then
         # Standard system
-        conda_tmp=$(mktemp -d)
+        # Create temp direcotry
+        mkdir -p "$scriptdir/tmpfs"
+        conda_tmp=$(mktemp -d --tmpdir="$scriptdir/tmpfs")
```

Comment:
I changed the temp directory is because when I test the `soconda.sh` I may terminate it in the middle, and it will leave `tmp.xxxx` directory in `/tmp`, but then there are other processes may also create similar `tmp.xxxx` directories. Then I don't know which one to delete.

I think we could add `tmpfs` to `.gitignore`. Also add `log_*` files.

Reply:
Ok, sounds good, lets keep all the temp files self-contained in the git tree.